### PR TITLE
Algodaily - Day 01

### PR DESCRIPTION
**Pull request by Shernz**
Prompt of finding the majority of numbers in an array

### CODE REVIEW

**Comment by Shernz** [comment]
In their solution, they used 2 pointer technique to find a number that is on both the halves

File: src/main/java/com/algodaily/technicalInterviewPrep/MajorityElement.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ */
+public class MajorityElement {
+    public static int majorityElement(int[] nums) {
+        // fill in
+        // with solution
+        int length = nums.length;
+        int majority = nums[0];
+        Map<Integer, Integer> numberMapper = new HashMap<>();
+        for(int num : nums) {
+            numberMapper.put(num, numberMapper.containsKey(num) ? numberMapper.get(num) + 1 : 1);
```
Comment (Xnkr) [inline src/main/java/com/algodaily/technicalInterviewPrep/MajorityElement.java:31]: Look into `computeIfPresent`

File: src/main/java/com/algodaily/technicalInterviewPrep/MajorityElement.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        }
+        for(Integer num : numberMapper.keySet()){
```
Comment (Xnkr) [inline src/main/java/com/algodaily/technicalInterviewPrep/MajorityElement.java:33]: Look into using `map.forEach` or `map.map` stream methods

**Review by Xnkr — APPROVED** [review]

File: src/main/java/com/algodaily/technicalInterviewPrep/MajorityElement.java (hunk 2 above)
Comment (Shernz) [reply src/main/java/com/algodaily/technicalInterviewPrep/MajorityElement.java:33]: I did think of using it but didn't know how to bring that into use.

File: src/main/java/com/algodaily/technicalInterviewPrep/MajorityElement.java (hunk 1 above)
Comment (Shernz) [reply src/main/java/com/algodaily/technicalInterviewPrep/MajorityElement.java:31]: Optional class??

Comment (Xnkr) [reply src/main/java/com/algodaily/technicalInterviewPrep/MajorityElement.java:31]: No https://www.geeksforgeeks.org/hashmap-computeifpresentkey-bifunction-method-in-java-with-examples/

**Pull request merged**